operator/src/main/java/oracle/kubernetes/operator/steps/HttpRequestProcessing.java

```diff
@@ -58,12 +58,14 @@ AuthorizationSource getAuthorizationSource() {
   }
 
   final HttpRequest.Builder createRequestBuilder(String url) {
-    return HttpRequest.newBuilder()
+    HttpRequest.Builder builder = HttpRequest.newBuilder()
           .uri(URI.create(url))
-          .header("Authorization", getAuthorizationSource().createBasicAuthorizationString())
           .header("Accept", "application/json")
           .header("Content-Type", "application/json")
           .header("X-Requested-By", "WebLogic Operator");
+    Optional.ofNullable(getAuthorizationSource())
```

Comment:
What problem led to this change? Is there a unit test for it?

Reply:
This change is also to avoid the NullPointerException when WebLogicCredentialsSecret is removed. I didn't know whether we need unit test when fixing the NPE. I'll look into adding unit test.